### PR TITLE
80 new source

### PR DESCRIPTION
**Pull request by davvarrr**
For today I offer you his sources if I'm motivated I'll see to add some tomorrow too :)

Some of the sources are the result of requests, so I've put in the relevant issues each time in order to close them once the request has been validated.

I hope the code is well formatted this time.

. Add new scrapper for Chapter on theme madara

"fr"
mangas-origines.xyz (unofficial)
mangas-origines.fr
scantrad-union.com
www1.bluesolo.org
pantheon-scan.com
hentai.scantrad-vf.cc
scan.hentai.menu
sushiscan.fr
www.lelmanga.com
shadowtrad.net

"en"
babelwuxia.com 
paragonscans.com -> #178
anshscans.org
cmreader.info
coffeemanga.io
coloredmanga.com
darkscans.com
lunarscan.org
reader.decadencescans.com
freemangatop.com
gdscans.com
freewebtooncoins.com
hentaimanga.me
hentaiwebtoon.com
hentaixcomic.com
hentaixyuri.com
hmanhwa.com
hscans.com

"es"
apollcomics.com
copypastescan.xyz
mundomanhwa.com
daprob.com
drakescans.com
emperorscan.com
dokkomanga.com
eromiau.com
mangareaderpro.com
herenscan.com

"tr"
mangawt.com 
webtoon-tr.com -> #62
mangakeyfi.net -> #58
ayatoon.com -> #59
mangasehri.com -> #52
webtoonhatti.com -> #61
guncelmanga.net
glorymanga.com
hayalistic.com

"ar"
galaxymanga.org -> #71
mangaatrend.net -> #65
azoranov.com
webtoonempire.com
swateam.me
gatemanga.com

"pt"
prismahentai.com -> #91
neoxscans.net -> #47
arthurscan.xyz
cerisescans.com
finalscans.com
fbsquads.com
cabaredowatame.site
nocsummer.com.br
foxwhite.com.br
gekkou.com.br
hentai.gekkouscans.com.br
gooffansub.com

"id"
worldmanhwas.bar -> #196
komiksay.site -> #201

"ru"
bestmanga.club

"vi"
hentaicube.net

### CODE REVIEW

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/mangareader/ar/SwaTeam.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		val nsfw = docs.selectFirst(".restrictcontainer") != null
+				|| docs.selectFirst(".info-right .alr") != null
+				|| docs.selectFirst(".postbody .alr") != null
+
+		return manga.copy(
+			description = docs.selectFirst("span.desc")?.html(),
+			state = mangaState,
+			author = author,
+			isNsfw = manga.isNsfw || nsfw,
+			tags = docs.select("div.spe a[rel*=tag]").mapNotNullToSet { a ->
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/mangareader/ar/SwaTeam.kt:87]:
```suggestion
			tags = docs.select("div.spe a[rel*=tag]").mapToSet { a ->
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/mangareader/fr/LelManga.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	MangaReaderParser(context, MangaSource.LELMANGA, pageSize = 21, searchPageSize = 20) {
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("www.lelmanga.com")
+
+	override val chapterDateFormat: SimpleDateFormat = SimpleDateFormat("MMMM d, yyyy", Locale.ENGLISH)
+
+	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+		val fullUrl = chapter.url.toAbsoluteUrl(domain)
+		val doc = webClient.httpGet(fullUrl).parseHtml()
+		val root = doc.body().selectFirst("div.maincontent")?.selectFirst("div#readerarea")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/mangareader/fr/LelManga.kt:32]:
```suggestion
		val root = doc.body().selectFirstOrThrow("div.maincontent").requireElementById("readerarea")
```

**Comment by davvarrr** [comment]
Thanks Koitharu, I've added opti to the 141bca5 commits.

**Comment by davvarrr** [comment]
Hello, it would be possible to close the issues concerned by the addition of this pr.

**Pull request merged**